lib/clustering/mean_shift.cpp

```diff
@@ -32,67 +32,54 @@ std::vector<Cluster> MeanShift::operator()(const PointCloud& cloud) const {
 	}
 
 	// Initialize some stuff used in the MeanShift loop
-	std::vector<Eigen::VectorXd> prev = points;
-	std::vector<Eigen::VectorXd> curr(points.size());
-	bool has_converged = false;
-  int iterations = 0; //for logging
-
-	// The MeanShift loop
-	while (!has_converged) {
-		iterations++;
-		if (iterations > MAX_ITERATIONS) {
-			BOOST_LOG_TRIVIAL(error) << "MeanShift: max number of iterations exceeded without convergence";
-			break;
-		}
-		// perform 1 iteration on each mode
-		for (int i=0; i<prev.size(); i++) {
-			curr[i] = iterate_mode(prev[i], prev);
-		}
-
-		// Merge step
-		// For every mode..
-		for (int i=0; i<curr.size(); i++) {
-			// Check modes we haven't already executed the (i)-loop for
-			for (int j=i+1; j<curr.size(); j++) {
-				Eigen::VectorXd diff = curr[i] - curr[j];
-
-				if (diff.norm() < MERGE_THRESHOLD) {
-					// Merge clusters. Erase one of the modes corresponding to the clusters and append points belonging to j to cluser of i
-					for (int k=0; k < clusters[j].points().size(); k++) {
-						clusters[i].points().push_back(clusters[j].points()[k]);
-					}
-					//BOOST_LOG_TRIVIAL(debug) << "Merging cluster " << j << " into cluster " << i << ", distance " << diff.norm();
-					clusters.erase(clusters.begin() + j);
-					curr.erase(curr.begin() + j);
-
-					break;
-				}
+	Eigen::VectorXd prevCenter;
+	std::vector<Eigen::VectorXd> currCenters = points;
+
+	// For each point...
+	for(PointIndex i = 0; i < clusters.size(); i++) {
+		prevCenter = points[i];
```

Comment:
looks like dead code, it's set again at line 47 but never read. (Probably you started with a while loop?)